File: my-solutions/0011-container-with-most-water/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int max_area = 0;

        while (left < right) {
            int min_line = min(height[left], height[right]);
            int current_area = min_line * (right - left);
            max_area = max(current_area, max_area);

            if (min_line == height[left]) {
                ++left;
            } else {
                --right;
            }
        }

        return max_area;
    }
};

File: my-solutions/0011-container-with-most-water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        start, end = 0, len(height) - 1
        area = 0

        while start < end:
            minHeight = min(height[start], height[end])
            area = max(area, (end - start) * minHeight)
            if minHeight == height[start]:
                start += 1
            else:
                end -= 1
        return area
